src/packages/user/user/workspace/components/user-workspace-profile-settings/user-workspace-profile-settings.element.ts

```diff
@@ -0,0 +1,138 @@
+import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { UUISelectElement } from '@umbraco-cms/backoffice/external/uui';
+import { UserResponseModel } from '@umbraco-cms/backoffice/backend-api';
+import { UMB_AUTH, UmbLoggedInUser } from '@umbraco-cms/backoffice/auth';
+import { firstValueFrom } from '@umbraco-cms/backoffice/external/rxjs';
+import { umbExtensionsRegistry } from '@umbraco-cms/backoffice/extension-registry';
+
+@customElement('umb-user-workspace-profile-settings')
+export class UmbUserWorkspaceProfileSettingsElement extends UmbLitElement {
+	@state()
+	private _user?: UserResponseModel;
+
+	@state()
+	private _currentUser?: UmbLoggedInUser;
+
+	@state()
+	private languages: Array<{ name: string; value: string; selected: boolean }> = [];
+
+	#authContext?: typeof UMB_AUTH.TYPE;
+	#userWorkspaceContext?: typeof UMB_USER_WORKSPACE_CONTEXT.TYPE;
+
+	constructor() {
+		super();
+
+		this.consumeContext(UMB_AUTH, (instance) => {
+			this.#authContext = instance;
+			this.#observeCurrentUser();
+		});
+
+		this.consumeContext(UMB_USER_WORKSPACE_CONTEXT, (instance) => {
+			this.#userWorkspaceContext = instance;
+			this.observe(this.#userWorkspaceContext.data, (user) => (this._user = user));
+		});
+	}
+
+	#onLanguageChange(event: Event) {
+		const target = event.composedPath()[0] as UUISelectElement;
+
+		if (typeof target?.value === 'string') {
+			this.#userWorkspaceContext?.updateProperty('languageIsoCode', target.value);
+		}
+	}
+
+	#observeCurrentUser() {
+		if (!this.#authContext) return;
+		this.observe(this.#authContext.currentUser, async (currentUser) => {
+			this._currentUser = currentUser;
+
+			if (!currentUser) {
+				return;
+			}
+
+			// Find all translations and make a unique list of iso codes
+			const translations = await firstValueFrom(umbExtensionsRegistry.extensionsOfType('localization'));
+
+			this.languages = translations
+				.filter((isoCode) => isoCode !== undefined)
+				.map((translation) => ({
+					value: translation.meta.culture.toLowerCase(),
+					name: translation.name,
+					selected: false,
+				}));
+
+			const currentUserLanguageCode = currentUser.languageIsoCode?.toLowerCase();
+
+			// Set the current user's language as selected
+			const currentUserLanguage = this.languages.find((language) => language.value === currentUserLanguageCode);
+
+			if (currentUserLanguage) {
+				currentUserLanguage.selected = true;
+			} else {
+				// If users language code did not fit any of the options. We will create an option that fits, named unknown.
+				// In this way the user can keep their choice though a given language was not present at this time.
+				this.languages.push({
+					value: currentUserLanguageCode ?? 'en-us',
+					name: currentUserLanguageCode ? `${currentUserLanguageCode} (unknown)` : 'Unknown',
+					selected: true,
+				});
+			}
+		});
```

Comment:
I know its annoying. But I havent found a better way jet. We need to give a unique string or Symbol to these observers so they replace each other if setup twice...  like this: `this.observe(..., ..., 'mySomethingObserver')`